diff -ur oasis-0.4.4-/src/plugins/ocamlbuild/MyOCamlbuildFindlib.ml oasis-0.4.4/src/plugins/ocamlbuild/MyOCamlbuildFindlib.ml
--- oasis-0.4.4-/src/plugins/ocamlbuild/MyOCamlbuildFindlib.ml
+++ oasis-0.4.4/src/plugins/ocamlbuild/MyOCamlbuildFindlib.ml
@@ -97,7 +97,7 @@
 
 (* This lists all supported packages. *)
 let find_packages () =
-  List.map before_space (split_nl & run_and_read "ocamlfind list")
+  List.map before_space (split_nl & run_and_read (exec_from_conf "ocamlfind" ^ " list"))
 
 
 (* Mock to list available syntaxes. *)
